### PR TITLE
Company registration and login

### LINKED ISSUE #64: Company Sign up and Sign in
Currently, we allow Guest users to register a Job on the platform. We need to restrict this for authenticated users.

Create a Sign in / Sign up flow for Companies and remove all company related fields from the new job page.

### PR DESCRIPTION
**Pull request by duduribeiro**
This is the very beginning part of the company dashboard.
This commit adds the company registration and login and requires login to
register a new job.

Closes #64 

image

image

# How to test

- Visit the root page
- Click on `Anunciar Vaga` button.
It should redirect to the login page

- Click on `Registrar` link, and set a new account.
It should create your account and login for you.

- Crate a new Job.
It should create a new job, redirect to the job page. The page should contain the company information (from the Registration page)

- Click on Logout
It should log out with success. 

You should visit the job show page without login.

### CODE REVIEW

File: app/assets/stylesheets/_login.scss [hunk 1]
```diff
@@ -0,0 +1,86 @@
+@import "variables";
+
+.other-links {
+  text-align: center;
+}
+
+.sign-box {
+  padding: 10px;
```
[the remark this answers is not in the document]
Comment (lastk) [reply app/assets/stylesheets/_login.scss]: life's to short for this :(

File: app/controllers/vacancies_controller.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 class VacanciesController < ApplicationController
+  before_action :require_login, except: %i[index show]
```
Comment (lastk) [inline app/controllers/vacancies_controller.rb]: its funny how i like the old way more, like: `except: [:index, :show]`

File: app/workers/setup_user_worker.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class SetupUserWorker
```
Comment (lastk) [inline app/workers/setup_user_worker.rb]: Why not `WelcomeUserWorker` ?

**Review by lastk — APPROVED** [review]

File: app/controllers/vacancies_controller.rb (hunk 2 above)
Comment (duduribeiro) [reply app/controllers/vacancies_controller.rb]: I prefer using %i. I can omit the `:`. And this is the styleguide

File: app/workers/setup_user_worker.rb (hunk 3 above)
Comment (duduribeiro) [reply app/workers/setup_user_worker.rb]: 👍

**Review by gentilfp — APPROVED** [review]

**Pull request merged**